lib/Transforms/ElementwiseToAffine/ElementwiseToAffine.cpp

```diff
@@ -130,10 +129,29 @@ struct ElementwiseToAffine
     RewritePatternSet patterns(context);
 
     patterns.add<ConvertAnyElementwiseMappableOpOnRankedTensors>(context);
-    target.markUnknownOpDynamicallyLegal([](Operation *op) {
+    target.markUnknownOpDynamicallyLegal([&](Operation *op) {
       // TODO (#768): to make this pass more widely applicable, it should take
       // a dialect and/or list of operations to restrict the conversion to
-      return !isElementwiseMappableOpOnRankedTensors(op);
+
+      auto isDialectInList =
+          llvm::any_of(restrictDialects, [&](std::string dialectName) {
+            return dialectName == op->getDialect()->getNamespace().str();
+          });
+      auto isOpInList = llvm::any_of(restrictOps, [&](std::string opName) {
+        return opName == op->getName().getStringRef().str();
+      });
+
+      auto is_EWMtoRT = isElementwiseMappableOpOnRankedTensors(op);
+
+      if (!restrictDialects.empty() && !restrictOps.empty()) {
+        return (isOpInList || isDialectInList) ? !is_EWMtoRT : true;
+      } else if (!restrictDialects.empty()) {
+        return isDialectInList ? !is_EWMtoRT : true;
+      } else if (!restrictOps.empty()) {
+        return isOpInList ? !is_EWMtoRT : true;
+      } else {
+        return !is_EWMtoRT;
+      }
```

Comment:
Rather than having this complex if/else here, I'd just set `isDialectInList` to true if there's no filter list specified, same for `isOpInList` (maybe rename them to `isDialectElegible` and `isOpElegibile` to make it less confusing), then you can simplify this to a single line:
```suggestion
        return (isDialectElegible || isOpElegibile) ? !is_EWMtoRT : true;
```

Reply:
Yeah, that much if-else there is complex and confusing there. 
But the test with no restriction 
`// RUN: heir-opt --convert-elementwise-to-affine %s | FileCheck --enable-var-scope %s`
fails with the simplification.

I find at least this much to be necessary
` if (restrictDialects.empty() && restrictOps.empty()) {
        return !is_EWMtoRT;
      }`
 `else {
        return (isDialectElegible || isOpElegible) ? !is_EWMtoRT : true;
      }`

or is there any other way to know that no restriction was provided and we can return the  `!is_EWMtoRT`

Reply:
> I find at least this much to be necessary

I think I agree, I'm struggling to work out the boolean logic in a simpler way (so I'd argue that if it's taking me this long to figure it out then what you have is more readable right now).

Reply:
My suggestion was to move the `empty` checks up into the same variable as the `contained` check:
```cpp
auto isDialectElegible =
    restrictDialects.empty() ||
    llvm::is_contained(restrictDialects,
                        op->getDialect()->getNamespace().str());
auto isOpElegible =
    restrictOps.empty() ||
    llvm::is_contained(restrictOps, op->getName().getStringRef().str());

auto is_EWMtoRT = isElementwiseMappableOpOnRankedTensors(op);

return (isDialectElegible && isOpElegible) ? !is_EWMtoRT : true;
 ```
I did have a logical error in my original suggestion, though, as it should be `&&` for the final line